workspaces/src/types/gas_meter.rs

```diff
@@ -0,0 +1,72 @@
+use std::sync::{Arc, Mutex};
+
+use crate::Worker;
+
+use super::Gas;
+use crate::result::Result;
+
+/// A hook that is called on every transaction that is sent to the network.
+/// This is useful for debugging purposes, or for tracking the amount of gas
+/// that is being used.
+pub type GasHook = Arc<Mutex<dyn FnMut(Gas) -> Result<()> + Send>>;
+
+/// Allows you to meter the amount of gas consumed by transaction(s).
+/// Note: This only works with parallel transactions that resolve to [`crate::Result::ExecutionFinalResult`]
+/// Example
+/// ```rust, ignore, no_run
+/// let mut worker = workspaces::sandbox().await?;
+/// let meter = GasMeter::now(&mut worker);
+///
+/// let wasm = std::fs::read(STATUS_MSG_WASM_FILEPATH)?;
+/// let contract = worker.dev_deploy(&wasm).await?;
+///
+/// contract
+///    .call("set_status")
+///    .args_json(json!({
+///        "message": "hello_world",
+///    }))
+///    .transact()
+///    .await?;
+///
+/// println!("Total Gas consumed: {}", meter.elapsed()?);
+/// ```
+pub struct GasMeter {
```

Comment:
What about the case where `GasMeter` is dropped? `Worker` now would be calling into gas hooks that are no longer alive or necessary

Reply:
I found channels work well for our use case instead of passing hooks. Acceptable?